### DISCUSSION TITLE
docker-compose openAM y openDJ not found

### DISCUSSION THREAD

**Discussion by prubilar** [OpenIdentityPlatform/OpenAM#595] [Q&A] [answered]
following the guide https://github.com/OpenIdentityPlatform/OpenAM/wiki/How-to-Start-OpenAM-and-OpenDJ-in-Separate-Docker-Contaners

It doesn't work, it gives the following errors:
1: disk does not mount for image openidentityplatform/opendj:latest
in the path /user/config/openam
2: opendj image gives following error

openidentityplatform-docker-opendj-1  | Configuring Directory Server ..... Stopping Directory Server ..... 
openidentityplatform-docker-opendj-1  | 
openidentityplatform-docker-opendj-1  | Error copying file /opt/opendj/template/config/MakeLDIF/addrate.template to
openidentityplatform-docker-opendj-1  | /opt/opendj/data/config/MakeLDIF/addrate.template.
openidentityplatform-docker-opendj-1  | See /tmp/opendj-setup-10683893455441672045.log for a detailed log of the
openidentityplatform-docker-opendj-1  | failed operation.
openidentityplatform-docker-opendj-1  | Please report this error and provide the log file mentioned above.
openidentityplatform-docker-opendj-1  | 
openidentityplatform-docker-opendj-1  | To see basic server configuration status and configuration, you can launch
openidentityplatform-docker-opendj-1  | /opt/opendj/bin/status
openidentityplatform-docker-opendj-1  | 
openidentityplatform-docker-opendj-1  | The Directory Server could not acquire an exclusive lock on file /opt/opendj/data/locks/server.lock: The attempt to obtain an exclusive lock on file /opt/opendj/data/locks/server.lock was rejected because the attempt to create the lock file failed: IOException(No such file or directory). This generally means that another instance of this server is already running
openidentityplatform-docker-opendj-1  | Loading initial schema:
openidentityplatform-docker-opendj-1  | Loading /opt/opendj/bootstrap/schema/99-user.ldif ...
openidentityplatform-docker-opendj-1  | The LDAP bind request failed: 91 (Connect Error)
openidentityplatform-docker-opendj-1  | Additional Information:  Connection refused
openidentityplatform-docker-opendj-1  | 
openidentityplatform-docker-opendj-1  | Unable to connect to the server at "localhost" on port 4444
openidentityplatform-docker-opendj-1  | 
openidentityplatform-docker-opendj-1  | Loading /opt/opendj/bootstrap/data/sample.ldif ...
openidentityplatform-docker-opendj-1  | The LDAP bind request failed: 91 (Connect Error)
openidentityplatform-docker-opendj-1  | Additional Information:  Connection refused
openidentityplatform-docker-opendj-1  | Try to upgrade OpenDJ
openidentityplatform-docker-opendj-1  | ls: cannot access '/opt/opendj/data/classes': No such file or directory
openidentityplatform-docker-opendj-1  | ./lib/_server-script.sh: 67: test: -a: unexpected operator
openidentityplatform-docker-opendj-1  | Starting OpenDJ
openidentityplatform-docker-opendj-1  | The Directory Server could not acquire an exclusive lock on file /opt/opendj/data/locks/server.lock: The attempt to obtain an exclusive lock on file /opt/opendj/data/locks/server.lock was rejected because the attempt to create the lock file failed: IOException(No such file or directory). This generally means that another instance of this server is already running
openidentityplatform-docker-opendj-1 exited with code 98

this my docker-compose
version: '3.9'

services:
  openam:
    image: openidentityplatform/openam:latest
    ports:
      - "8080:8080"
    hostname: openam.example.com
    volumes:
      - ./persistence/openam:/usr/openam/config
    depends_on:
      - opendj

  opendj:
    image: openidentityplatform/opendj:latest
    hostname: opendj.example.com
    ports:
      - "1389:1389"
      - "1636:1636"
      - "4444:4444"
    volumes:
      - ./opendj/bootstrap/data/:/opt/opendj/bootstrap/data
      - ./opendj/bootstrap/schema/:/opt/opendj/bootstrap/schema
      - ./persistence/opendj:/opt/opendj/data
    environment:
      - BASE_DN=dc=openam,dc=openidentityplatform,dc=org

**Comment by maximthomas** [comment, marked as answer]
Seems there is an issue with user rights between the host machine and Docker container. Try to add `user: root` do the `docker-compose.yml` file and try again:

```yml
  opendj:
    image: openidentityplatform/opendj:latest
    hostname: opendj.example.com
    user: root
    ports:
      - "1389:1389"
      - "1636:1636"
      - "4444:4444"
    volumes:
      - ./opendj/bootstrap/data/:/opt/opendj/bootstrap/data
      - ./opendj/bootstrap/schema/:/opt/opendj/bootstrap/schema
      - ./persistence/opendj:/opt/opendj/data
    environment:
      - BASE_DN=dc=openam,dc=openidentityplatform,dc=org
```

**Comment by prubilar** [comment]
Thanks.
